src/nms-api/IMessageConsumer.cs

```diff
@@ -69,10 +75,15 @@ public interface IMessageConsumer : System.IDisposable
         IMessage ReceiveNoWait();
 
         /// <summary>
-        /// An asynchronous listener which can be used to consume messages asynchronously
```

Comment:
This is an async consumer, its callback style rather than task based. Not sure why change of doc.

Reply:
The message delivery here itself though is async callback. Its still async in delivery style rather than the poll model on receive.

Reply:
It's not the matter of callback or poll. It's just a matter or naming thing properly so other developers can understand what's going on here.

Reply:
Here's an example how it's implemented in RabbitMQ .NET Client:
This is what I'm trying to add --> https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/7.x/projects/RabbitMQ.Client/client/events/AsyncEventingBasicConsumer.cs 
This is what we have with current API --> https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/7.x/projects/RabbitMQ.Client/client/events/EventingBasicConsumer.cs

Reply:
imho its useful change